docs/output-targets/docs-json.md

```diff
@@ -30,80 +37,179 @@ export const config: Config = {
 };
 ```
 
-Check out the typescript declarations for the JSON output: https://github.com/ionic-team/stencil/blob/main/src/declarations/stencil-public-docs.ts
+The JSON file output by Stencil conforms to the [`JsonDocs` interface in
+Stencil's public TypeScript
+declarations](https://github.com/ionic-team/stencil/blob/main/src/declarations/stencil-public-docs.ts).
 
 
 ## CSS Variables
 
-Stencil will also document CSS variables when you specify them via jsdoc-style comments inside your css/scss files:
+Stencil can document CSS variables if you annotate them with JSDoc-style
+comments in your css/scss files. If, for instance, you had a component with a
+css file like the following:
+
+```css title="src/components/my-button/my-button.css"
+:host {
+  /**
+   * @prop --background: Background of the button
+   * @prop --background-activated: Background of the button when activated
+   * @prop --background-focused: Background of the button when focused
+   */
+  --background: pink;
+  --background-activated: aqua;
+  --background-focused: fuchsia;
+}
+```
 
-```css
-/**
- * @prop --background: Background of the button
- * @prop --background-activated: Background of the button when activated
- * @prop --background-focused: Background of the button when focused
- */
+Then you'd get the following in the JSON output:
+
+```json
+"styles": [
+  {
+    "name": "--background",
+    "annotation": "prop",
+    "docs": "Background of the button"
+  },
+  {
+    "name": "--background-activated",
+    "annotation": "prop",
+    "docs": "Background of the button when activated"
+  },
+  {
+    "name": "--background-focused",
+    "annotation": "prop",
+    "docs": "Background of the button when focused"
+  }
+]
 ```
 
+:::note
+This functionality works with both standard CSS and with Sass, although for the
+latter you'll need to have the
+[@stencil/sass](https://github.com/ionic-team/stencil-sass) plugin installed
+and configured.
+:::
 
 ## Slots
 
-Slots can be documented by adding `@slot` tags to the doc comments above the `@Component` decorator.
+If one of your Stencil components makes use of
+[slots](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot) for
+rendering children you can document them by using the `@slot` JSDoc tag in the
+component's comment.
+
+For instance, if you had a `my-button` component with a slot you might document
+it like so:
+
+```tsx title="src/components/my-button/my-button.tsx"
+import { Component, h } from '@stencil/core';
 
-```tsx
 /**
- * @slot slotName - slotDescription
  * @slot buttonContent - Slot for the content of the button
  */
- 
- @Component({
-  tag: '...'
-}) ...
+@Component({
+  tag: 'my-button',
+  styleUrl: 'my-button.css',
+  shadow: true,
+})
+export class MyButton {
+  render() {
+    return <button><slot name="buttonContent"></slot></button>
+  }
+}
 ```
 
+This would show up in the generated JSON file like so:
+
+```json
+"slots": {
+  "name": "buttonContent",
```

Comment:
Do you need to specify a name in the JSDoc? Like, does this work with a default/unnamed slot?

Reply:
I'll check, I'm not sure

Reply:
So it looks like it just works based on the `@slot` tag in the JSDoc, I don't believe it does any check to see whether slots are actually present in the JSX for the component, just pulls in the `@slot` tag whether or not it has a name